### PR TITLE
PRO-2051 Adds a spinner to the install process

### LINKED ISSUE #13: Create command does not complete with npm 7
## To Reproduce

1. Have npm 7 installed on your local machine
2. install the CLI tool
3. `apos create newone`

## Expected behavior
It should install the A3 boilerplate and create the first user

## Describe the bug
Terminal output stops after step 3:

```bash
 ✘  ~/Sites  apos create newone

 Apostrophe  create  Grabbing the starter from Github [1/4]
Cloning into 'newone'...

 Apostrophe  create  Adding your project shortname (newone) [2/4]

 Apostrophe  create  Installing packages [3/4]
rm: no such file or directory: package-lock.json

rm: no such file or directory: yarn.lock

^C
```

## Details

**Version of Node.js:**
Node 14, npm 7.6.1

### PR DESCRIPTION
**Pull request by abea**
Resolves #13

### CODE REVIEW

**Review by boutell — CHANGES_REQUESTED** [review]
I think this is correct, but it is also very hard to follow.

So I suggest that you write a spawnWithSpinner function which returns a promise so it can be awaited normally in the flow of the code.

Within that function, you'd invoke the resolve function of the promise on depChild.on('exit') after calling spinner.stop().

**Comment by boutell** [comment]
You should also invoke `reject` if the exit status is no good.

File: lib/commands/create.js [hunk 1]
```diff
@@ -98,3 +110,36 @@ module.exports = function (program) {
...
+        shell: true
+      });
+
+      spinner.start();
+
+      installer.on('exit', (code) => {
+        spinner.stop();
+
+        if (code !== null && code !== 0) {
+          reject(new Error(`Install spawn process error with code ${code}`));
```
Comment (boutell) [inline lib/commands/create.js:134]: I think it's a good idea to "return reject(...)", "return resolve(...)", etc. You successfully avoided any inadvertent fallthroughs here but it's a good pattern to make sure that remains the case.

File: lib/commands/create.js [hunk 2]
```diff
@@ -98,3 +110,36 @@ module.exports = function (program) {
 function replaceInConfig(regex, replacement) {
   util.replaceInFiles([ './app.js', './package.json' ], regex, replacement);
 }
+
+async function spanWithSpinner () {
```
Comment (boutell) [inline lib/commands/create.js:114]: Shouldn't this be spawnWithSpinner? And passing the command and its text to it so you can reuse it for other commands would make sense, I think.

On a way less technical note, I would suggest including "this may take a while" in the message. Just because some users (cough me cough) will still suspect the worst even with a spinner and do better with an affirmative confirmation that yes, this is gonna take a minute and not say much in the meantime.

File: lib/commands/create.js [hunk 3]
```diff
@@ -98,3 +110,36 @@ module.exports = function (program) {
...
+  const spinner = ora({
+    text: 'Installing packages...',
+    // stream: process.stdout,
+    interval: 100,
+    isEnabled: true
+  });
+
+  return new Promise((resolve, reject) => {
```
Comment (boutell) [inline lib/commands/create.js:122]: Great job here on the promisification, just one note see below.

**Review by boutell — CHANGES_REQUESTED** [review]

File: lib/commands/create.js (hunk 2 above)
Comment (abea) [reply lib/commands/create.js:114]: Typo goof.

File: lib/util.js [hunk 4]
```diff
@@ -145,6 +147,43 @@ util.secret = function() {
...
+  const spinner = ora({
+    text: options.spinnerMessage,
+    interval: 100,
+    isEnabled: true
+  });
+
+  return new Promise((resolve, reject) => {
+    try {
+      const installer = spawn(command, {
+        shell: true
```
Comment (boutell) [inline lib/util.js:167]: Sorry, I should have asked this before: will the user ever see the output? That's especially important if there's an error.

**Review by boutell — CHANGES_REQUESTED** [review]

File: lib/util.js (hunk 4 above)
Comment (abea) [reply lib/util.js:167]: Yes, it's logged here https://github.com/apostrophecms/cli/pull/14/files/4d6bbfb8f4bf4b5d76712784a2da911d255ff76a..52586ebc5934eba202a9e09c74975ed09699027f#diff-98638a4ebf5a3e8155511395d90744a4e069d1a55960da5a9d4c25dc51d07602L75

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (abea) [inline .vscode/settings.json:1]: Remove this.

**Comment by boutell** [comment]
To keep this convo moving: my concern is that while we've hidden the output in the middle of the execution of the command in favor of a spinner, we do need to show the actual output at the end, *especially* if the exit code is nonzero. And the exit code is not sufficient information to share. They need to know what npm said.

This involves capturing stdout and stderr, which isn't too bad:

```
let output = '';
installer.stdout.on('data', (data) => {
  output += data.toString();
});

installer.stderr.on('data', (data) => {
  output += data.toString();
});
// now on exit you can log output
```

Sorry to be a pain about this, but we do need to know the real reason when an error occurs.

**Review by boutell — APPROVED** [review]
